distribution/pom.xml

```diff
@@ -108,6 +108,8 @@
                                 <argument>io.druid.extensions:druid-stats</argument>
                                 <argument>-c</argument>
                                 <argument>io.druid.extensions:druid-examples</argument>
+                                <argument>-c</argument>
+                                <argument>io.druid.extensions:sample-client-sslcontext</argument>
```

Comment:
I guess it can be used for simple use cases. This extension can serve as an example for users for how to inject their own `SSLContext` if this extension does not satisfy their needs. 
@nishantmonu51 @gianm I need advice on this next point  -
Another motivation to implement it as an extension was to let the users understand what they are doing. For example users can point HTTPClient's `trustStore` to the same `keyStore` which server uses for storing its signed certificate and possibly some private keys. So, is it OK to do this ? I don't know so let the users decide and understand what they are doing. Does this reasoning make sense ?